server/channels/api4/preference.go

```diff
@@ -122,7 +122,13 @@ func updatePreferences(c *Context, w http.ResponseWriter, r *http.Request) {
 				return
 			}
 
-			if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), post.ChannelId, model.PermissionReadChannelContent) {
+			channel, err := c.App.GetChannel(c.AppContext, post.ChannelId)
```

Comment:
We may want to revisit this. You are getting a channel for each flagged post, which may have performance implications. A "not great but better" solution would be to keep a map of channels that you already checked whether you have permissions, so you don't have to do this several times for several flagged posts in the same channel.

Not sure what is the maximum number of flagged posts we can have, and maybe this is not an issue here because of that. 0/5

Nevertheless, please take another look to the changes to see if we are doing this in any loop that may have a similar performance concern.

Reply:
Yeah I guess we could add the channel mapping concept here as well to be safe, though I don't think there would be as much of a concern here vs. some of the other APIs since the number of flagged posts likely wouldn't be high, and we are already getting each post one by one here. I guess best not to make it worse :P